src/Workspaces/Core/Portable/FindSymbols/SymbolFinder_Hierarchy.cs

```diff
@@ -184,405 +213,188 @@ public static partial class SymbolFinder
+        public static Task<IEnumerable<INamedTypeSymbol>> FindImplementationsAsync(
+            INamedTypeSymbol type, Solution solution, IImmutableSet<Project> projects = null, CancellationToken cancellationToken = default)
+#pragma warning restore RS0026 // Do not add multiple public overloads with optional parameters
         {
-            if (symbolToMatch == null)
-            {
-                return false;
-            }
-
-            if (OriginalSymbolsMatchCore(searchSymbol, symbolToMatch, solution, searchSymbolCompilation, symbolToMatchCompilation, cancellationToken))
-            {
-                return true;
-            }
-
-            if (searchSymbol.Kind == SymbolKind.Namespace && symbolToMatch.Kind == SymbolKind.Namespace)
-            {
-                // if one of them is a merged namespace symbol and other one is its constituent namespace symbol, they are equivalent.
-                var namespace1 = (INamespaceSymbol)searchSymbol;
-                var namespace2 = (INamespaceSymbol)symbolToMatch;
-                var namespace1Count = namespace1.ConstituentNamespaces.Length;
-                var namespace2Count = namespace2.ConstituentNamespaces.Length;
-                if (namespace1Count != namespace2Count)
-                {
-                    if ((namespace1Count > 1 &&
-                         namespace1.ConstituentNamespaces.Any(n => NamespaceSymbolsMatch(n, namespace2, solution, cancellationToken))) ||
-                        (namespace2Count > 1 &&
-                         namespace2.ConstituentNamespaces.Any(n2 => NamespaceSymbolsMatch(namespace1, n2, solution, cancellationToken))))
-                    {
-                        return true;
-                    }
-                }
-            }
-
-            return false;
+            return FindImplementationsAsync(type, solution, transitive: true, projects, cancellationToken);
         }
 
-        private static bool OriginalSymbolsMatchCore(
-            ISymbol searchSymbol,
-            ISymbol symbolToMatch,
-            Solution solution,
-            Compilation searchSymbolCompilation,
-            Compilation symbolToMatchCompilation,
-            CancellationToken cancellationToken)
+        /// <summary>
+        /// Finds the accessible <see langword="class"/> or <see langword="struct"/> types that implement the given
+        /// interface.
+        /// </summary>
+        /// <param name="type">The symbol to find derived types of.</param>
+        /// <param name="solution">The solution to search in.</param>
+        /// <param name="transitive">If the search should stop at immediately derived interfaces, or should continue past that.</param>
+        /// <param name="projects">The projects to search. Can be null to search the entire solution.</param>
+#pragma warning disable RS0026 // Do not add multiple public overloads with optional parameters
```

Comment:
the problem is that this means we can't have overloads (since we have optional cancellation tokens).  so we effectively disregard this warning everywhere.  i think the goal was to prevent the cases where overloads could prevent the right method being called.  but that oesn't happen here with these parameter types.